src/components/NumberInput/NumberInput.cy.tsx

```diff
@@ -0,0 +1,144 @@
+        cy.get(NUMBER_INPUT_ELEMENT).type('{shift}{leftarrow}');
+        cy.get(NUMBER_INPUT_ELEMENT).should('have.value', '-10');
+    });
+
+    it('should decrease value by 1 on arrow down', () => {
+        cy.mount(<NumberInput />);
+        cy.get(NUMBER_INPUT_ELEMENT).type('{downarrow}');
+        cy.get(NUMBER_INPUT_ELEMENT).should('have.value', '-1');
+    });
+
+    it('should decrease value by 10 on shift + arrow down', () => {
+        cy.mount(<NumberInput />);
+        cy.get(NUMBER_INPUT_ELEMENT).type('{shift}{downarrow}');
+        cy.get(NUMBER_INPUT_ELEMENT).should('have.value', '-10');
+    });
+
+    it('should render increment buttons', () => {
+        cy.mount(<NumberInput incrementable />);
+        cy.get(NUMBER_INPUT_DECREMENT).should('be.visible');
+        cy.get(NUMBER_INPUT_INCREMENT).should('be.visible');
+    });
+
+    it('should increase value by 1 on increment button click', () => {
+        cy.mount(<NumberInput incrementable />);
+        cy.get(NUMBER_INPUT_INCREMENT).click();
+        cy.get(NUMBER_INPUT_ELEMENT).should('have.value', '1');
+    });
+
+    it('should increase value by 10 on increment button shift + click', () => {
+        cy.mount(<NumberInput incrementable />);
+        cy.get(NUMBER_INPUT_INCREMENT).click({ shiftKey: true });
+        cy.get(NUMBER_INPUT_ELEMENT).should('have.value', '10');
+    });
+
+    it('should decrease value by 1 on decrement button click', () => {
+        cy.mount(<NumberInput incrementable />);
+        cy.get(NUMBER_INPUT_DECREMENT).click();
+        cy.get(NUMBER_INPUT_ELEMENT).should('have.value', '-1');
+    });
+
+    it('should decrease value by 10 on decrement button shift + click', () => {
+        cy.mount(<NumberInput incrementable />);
+        cy.get(NUMBER_INPUT_DECREMENT).click({ shiftKey: true });
+        cy.get(NUMBER_INPUT_ELEMENT).should('have.value', '-10');
+    });
+
+    it('should sequential increase value on shift + increment button hold', () => {
+        cy.mount(<NumberInput incrementable />);
+        cy.get(NUMBER_INPUT_INCREMENT).trigger('keydown', { keyCode: 16, release: false });
+        cy.get(NUMBER_INPUT_INCREMENT).trigger('mousedown', { release: false });
+        cy.wait(2000);
+        cy.get(NUMBER_INPUT_INCREMENT).trigger('mouseup');
+        cy.get(NUMBER_INPUT_INCREMENT).trigger('keyup', { keyCode: 16 });
+        cy.get(NUMBER_INPUT_ELEMENT).should('have.value', '8');
+    });
+
+    it('should sequential decrease value on shift + decrement button hold', () => {
+        cy.mount(<NumberInput incrementable />);
+        cy.get(NUMBER_INPUT_DECREMENT).trigger('keydown', { keyCode: 16, release: false });
+        cy.get(NUMBER_INPUT_DECREMENT).trigger('mousedown', { release: false });
+        cy.wait(2000);
+        cy.get(NUMBER_INPUT_DECREMENT).trigger('mouseup');
+        cy.get(NUMBER_INPUT_DECREMENT).trigger('keyup', { keyCode: 16 });
+        cy.get(NUMBER_INPUT_ELEMENT).should('have.value', '-8');
```

Comment:
To reduce risk of flakeyness, wdyt about replacing `'have.value', '-8'` by `'be.within', -7, -9`?